portfolio/src/main/java/com/google/sps/classes/UtilityClass.java

```diff
@@ -0,0 +1,42 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.classes;
+
+import com.google.gson.Gson;
+import com.google.sps.classes.UserAuthentication;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * User object created to group login/logout url and status.
+ */
+public final class UtilityClass {
+  /**
+   * Converts ArrayList of comments to JSON using GSON class.
+   */
+  public static String convertToJsonUsingGson(List<String> comments) {
+    Gson gson = new Gson();
+    return gson.toJson(comments);
+  }
+
+  /**
+   * Overloaded convertToJsonUsingGson() function that converts user object 
+   * to JSON using GSON class.
+   */
+  public static String convertToJsonUsingGson(UserAuthentication user) {
+    Gson gson = new Gson();
+    return gson.toJson(user);
+  }
```

Comment:
I believe you may be able to actually combine these by doing something like this: 

```suggestion
  /**
   * Overloaded convertToJsonUsingGson() function that converts an object 
   * to JSON using the GSON class.
   */
  public static String convertToJsonUsingGson(Object object) {
    Gson gson = new Gson();
    return gson.toJson(object);
  }
```

The toJson method takes in an object (https://www.javadoc.io/doc/com.google.code.gson/gson/latest/com.google.gson/com/google/gson/Gson.html) so I think you should be able to just use Object rather than having individual converToJsonUsingGson methods for every type of input.  Try changing it to take in Object and see if all of your use cases can call the same method! If you get errors for type mismatches, let me know!

Reply:
Didn't know of the existence of this Object class but it does make things easier. Thanks!